docs/how-xtrabackup-works.md

```diff
@@ -1,123 +1,149 @@
+starts, the application inspects the data files and the transaction log and does the following:
+
+* Applies committed transaction log entries to the data files
+
+* Removes uncommitted transactions - any change made within those transactions is discarded
+
+## Version changes
+
+[Percona XtraBackup 8.0.30-23](release-notes/8.0/8.0.30-23.0.md) adds 
+the [--register-redo-log-consumer] parameter. This parameter is 
+disabled by default. When enabled, the parameter lets Percona XtraBackup 
+register as a redo log consumer at the start of the backup. The server 
+does not remove a redo log that Percona XtraBackup (the consumer) 
+has not yet copied. The consumer reads the redo log and manually 
+advances the log sequence number (LSN). The server blocks the writes 
+during the process. The server determines when to purge the log based on the redo log consumption.
+
+
+Percona XtraBackup works by remembering the LSN when it starts and then 
+copying away the data files. It takes some time to do this, so if the files change, they reflect the 
+state of the database at different points. At the same time, Percona XtraBackup runs a background 
+process that watches the transaction log files and copies changes from them. 
+Percona XtraBackup does this continually because the
+transaction logs are written round-robin and can be reused. Percona XtraBackup needs the transaction log record for every change to the data files since it began execution.
+
+## Locks
+
+Locks are used in a backup to ensure data consistency. You cannot insert new data or modify data during the backup. Locking can cause performance issues, especially for frequently accessed tables.
+
+Percona XtraBackup uses `LOCK INSTANCE FOR BACKUP` to take a backup log for the entire process, including InnoDB and non-InnoDB data. The `xtrabackup` binary copies the InnoDB data and then copies the MyISAM tables and files. 
+
+The `xtrabackup` binary backs up the following file types:
+
+* .FRM
+
+* .MRG
+
+* .MYD
+
+* .MYI
+
+* .CSM
+
+* .CSV
+
+* .SDI
+
+* .PAR 
+
+### LOCK INSTANCE FOR BACKUP
+
+The `LOCK INSTANCE FOR BACKUP` statement is a feature introduced in MySQL 8.0 that allows you to acquire an instance-level backup lock. Executing the `LOCK INSTANCE FOR BACKUP` statement prevents any other session from acquiring a backup lock at the instance level. It ensures no changes to the data files while the backup process is running. This statement is executed after backing up all InnoDB/XtraDB data and logs and locks all non-InnoDB tables to safely copy non-InnoDB data without blocking any Data Manipulation Language (DML)  queries that modify InnoDB tables. The `LOCK INSTANCE FOR BACKUP` statement provides a more efficient and targeted way to lock the necessary tables during the backup process, reducing the impact on the database's availability and performance.
```

Comment:
I think we should also mention about LTFB used on Percona Server. This is because LIFB doesn't prevent DML operations on non-innodb tables like MyISAM. This can corrupt myisam tables. So we different mechanism to prevent corruption of myisam tables.

So against Percona Server
1. LIFB
2. start redo copy thread
3. copy InnoDB data files
4. after InnoDB data file is over, copy non-innodb tables
5. after file copy is over, ask PFS log_status about checkpoint_lsn, write_lsn, binlog position
6. stop redo copy thread at the write_lsn from PFS
7. write binlog info, gtid info xtrabackup_binlog_info file
8. release backup lock

against upstream:
1. LIFB
2. start redo copy thread
3. copy InnoDB data files
4. after InnoDB data file is over,  Take FTWRL 
5. copy non-innodb tables
6. release FTWRL
7. after file copy is over, ask PFS log_status about checkpoint_lsn, write_lsn, binlog position
8. stop redo copy thread at the write_lsn from PFS
9. write binlog info, gtid info xtrabackup_binlog_info file
10. release backup lock

Reply:
LIFB-> Lock Instance for Backup (available on both upstream and Percona server for MySQL)
LTFB -> Lock Tables for Backup (only in PS)